syntax = 'proto3';

package gg.protobuf;

message Function {
  string exe = 1;
  repeated string args = 2;
  string hash = 3;
  repeated string envars = 4;
}

message InFile {
  string filename = 1;
  string hash = 2;
  uint32 order = 3;
  uint64 size = 4;

  enum Type {
    FILE = 0;
    EXECUTABLE = 1;
    DUMMY_DIRECTORY = 2;
  }

  Type type = 5;
}

message Thunk {
  Function function = 1;
  repeated InFile infiles = 2;
  string outfile = 3;
}

message LambdaInFile {
  string hash = 1;
  uint64 size = 3;
  bool executable = 2;
}

message LambdaEvent {
  string thunk_hash = 1;
  string s3_bucket = 2;
  repeated LambdaInFile infiles = 3;
  string thunk_data = 4;
}

message RemoteResponse {
  string thunk_hash = 1;
  string output_hash = 2;
  uint64 output_size = 3;
  bool executable_output = 4;
}
